core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -882,14 +900,85 @@ public void testCounterDecrementedIfShardOperationThrowsException() throws Inter
         assertPhase(task, "failed");
     }
 
+    public void testReroutePhaseRetriedAfterDemotedPrimary() {
```

Comment:
I meant removing this in favor of  what I thought would be a centralized reroute test , but I know see there is no suche thing and we do it on a case by case basis. So Ignore me, all ok.